src/specialized/aarch64.rs

```diff
@@ -19,7 +19,7 @@ impl State {
 
     #[cfg(feature = "std")]
     pub fn new(state: u32) -> Option<Self> {
-        if std::arch::is_aarch64_feature_detected!("crc") {
+        if cfg!(target_feature = "crc") && std::arch::is_aarch64_feature_detected!("crc") {
```

Comment:
The point of the `is_*_feature_detected!()` is to allow using crc instructions even when an executable was built without it explicitly enabled. Note that `is_*_feature_detected!()` already internally checks `cfg!(target_feature = "...")` and returns true no matter what gets detected at runtime.

The respective intrinsics should just be implemented in cg_clif.

Reply:
> The point of the `is_*_feature_detected!()` is to allow using crc instructions even when an executable was built without it explicitly enabled. Note that `is_*_feature_detected!()` already internally checks `cfg!(target_feature = "...")` and returns true no matter what gets detected at runtime.

Yes, of course. There's no way to tell if a target feature was simply forgotten or explicitly disabled, though. (Cargo features have this problem too.)

> The respective intrinsics should just be implemented in cg_clif.

I agree they should. I haven't done that myself yet so it may take a while.

---

If we don't want to assume a missing target feature is explicit disablement, we can close this PR. It's easy for me to just patch crc32fast with Cargo until Cranelift gains support.